### PR TITLE
Add vehicle damage tracking to Core vehicle 

### PR DESCRIPTION
**Pull request by dexslab**
Needs for proper damage calculations for weapons and vehicle degradation.

### CODE REVIEW

**Comment by dexslab** [comment]
I added more to my PR please review again thanks 😄

File: src/core/server/extensions/vehicleFuncs.ts [hunk 1]
```diff
@@ -344,6 +344,13 @@ export default class VehicleFuncs {
             vehicle.engineHealth = vehicle.data.engineHealth;
         }
 
+        if (vehicle.data.damage) {
+            vehicle.setDamageStatusBase64(vehicle.data.damage);
+        }
+        if (vehicle.data.health) {
+            vehicle.setHealthDataBase64(vehicle.data.damage);
```
[the remark this answers is not in the document]
Comment (dexslab) [reply src/core/server/extensions/vehicleFuncs.ts:351]: I am in the process of revising this please wait for next commit 😄

File: src/core/server/extensions/vehicleFuncs.ts [hunk 2]
```diff
@@ -1260,4 +1267,139 @@ export default class VehicleFuncs {
...
+                    damageLevel: vehicle.getBumperDamageLevel(0),
+                },
+                'RearBumper': {
+                    damageLevel: vehicle.getBumperDamageLevel(1),
+                }
+            }
+
+        }
+        return damageObject;
+    }
```
Comment (Derison) [inline src/core/server/extensions/vehicleFuncs.ts:1331]:
```suggestion
            }
        };
    }
```

**Review by Derison — CHANGES_REQUESTED** [review]

File: src/core/server/extensions/vehicleFuncs.ts [hunk 3]
```diff
@@ -1260,4 +1267,139 @@ export default class VehicleFuncs {
             vehicle.data.tuning.driftModeEnabled = enabled;
         }
     }
+
+    static getDamage(vehicle: alt.Vehicle): object {
+        let damageObject = {
```
[the remark this answers is not in the document]
Comment (dexslab) [reply src/core/server/extensions/vehicleFuncs.ts:1272]: Ya prolly

**Pull request merged**